src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageDirectiveFeature.cs

```diff
@@ -50,20 +50,16 @@ public static bool TryGetPageDirective(ILogger logger, RazorProjectItem projectI
             var documentIRNode = codeDocument.GetDocumentIntermediateNode();
             if (PageDirective.TryGetPageDirective(documentIRNode, out var pageDirective))
             {
+                if (pageDirective.DirectiveNode is MalformedDirectiveIntermediateNode malformedNode)
+                {
+                    logger.MalformedPageDirective(sourceDocument.FilePath, malformedNode.Diagnostics);
+                }
+
                 template = pageDirective.RouteTemplate;
```

Comment:
Presumably this doesn't affect tests (Travis \ AppV are failing because the Razor changes are in).